### PR TITLE
WishList Page

### PR DESCRIPTION
**Pull request by prajapati-priyanka**
-Add Wishlist management
-User can add and remove products from wishlist page.
-Authentication is not implemented, so any user can visit wishlist page.

### CODE REVIEW

File: src/context/product-context.js [hunk 1]
```diff
@@ -1,52 +1,59 @@
...
-const finalfilteredList = filterRating(filteredCategoryData, filterState.sortByRating)
-// console.log("in context", sortedData)
-
-    return <ProductContext.Provider value={{filterState, finalfilteredList, filterDispatch}}>
-        {children}
+  useEffect(() => {
+    (async () => {
+      try {
+        const res = await axios.get("./api/products");
+        console.log("data from api", res.data.products);
```
Comment (AnkurChunekar) [inline src/context/product-context.js]: `console.log`s should be removed if they are no longer needed.

File: src/context/product-context.js [hunk 2]
```diff
@@ -1,52 +1,59 @@
...
+
+  const sortedData = sortPrice(filterState.dataList, filterState.sortBy);
+  const filteredData = filteredProduct(sortedData, filterState);
+  const filteredPriceData = filterPrice(filteredData, filterState.priceValue);
+  const filteredCategoryData = filterCategory(filteredPriceData, filterState);
+  const finalfilteredList = filterRating(
+    filteredCategoryData,
+    filterState.sortByRating
+  );
+  // console.log("in context", sortedData)
```
Comment (AnkurChunekar) [inline src/context/product-context.js]: Remove unnecessary comments

**Comment by AnkurChunekar** [comment]
priyanka pr
If I click on the like button more than once by mistake, the same product gets added, you can disable the button until the request is complete.

File: src/reducer/wishList-reducer.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      return {
+        ...state,
+        wishListItem: [...action.payload],
+        wishListCount: state.wishListCount + 1,
+      };
+    case "DELETE_PRODUCT_FROM_WISHLIST":
+      return {
+        ...state,
+        wishListItem: [...action.payload],
+        wishListCount: state.wishListCount - 1,
```
Comment (AnkurChunekar) [inline src/reducer/wishList-reducer.js]: IMO, you `wishlistCount` can be easily calculated from `wishListItem.length` method, there's no need for having a separate property for count in the state. I did it the same way at first but later i realised, that in some cases, it creates more bugs.

File: src/context/wishList-context.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    try {
+      const response = await axios.delete(
+        `/api/user/wishlist/${products._id}`,
+        config
+      );
+
+      wishListDispatch({
+        type: "DELETE_PRODUCT_FROM_WISHLIST",
+        payload: response.data.wishlist,
+      });
```
Comment (AnkurChunekar) [inline src/context/wishList-context.js]: check for the response status before updating the state , you havent checked if the request is successfull or not

**Comment by AnkurChunekar** [comment]
All the bugs have been fixed, and functionality works fine. Great work @prajapati-priyanka

**Review by AnkurChunekar — APPROVED** [review]

File: src/components/footer/Footer.css [hunk 5]
```diff
@@ -1,6 +1,6 @@
 .footer-content {
-    bottom: 0;
-    margin: 2rem 0 0 0;
+    /* bottom: 0; */
+    margin: 4rem 0 0 0;
```
Comment (mariyasada) [inline src/components/footer/Footer.css]: remove commented code

File: src/context/wishList-context.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        config
+      );
+
+      if (response.status === 201) {
+        wishListDispatch({
+          type: "ADD_PRODUCT_TO_WISHLIST",
+          payload: response.data.wishlist,
+        });
+        // setIsDisabled(false)
+      } else {
```
Comment (mariyasada) [inline src/context/wishList-context.js]: remove this

**Review by mariyasada — APPROVED** [review]
the code structure is good. good to merge

**Pull request merged**